actors/builtin/miner/miner_state.go

```diff
@@ -126,6 +126,7 @@ type SectorPreCommitInfo struct {
 	SealRandEpoch abi.ChainEpoch
 	DealIDs       []abi.DealID
 	Expiration    abi.ChainEpoch
+	ReplaceSector abi.SectorNumber // "Committed capacity" no-deal sector to replace, valid only with non-empty deals
```

Comment:
`0` is a valid `SectorNumber`, this should probably be a pointer

Reply:
Elsewhere in this PR I have defined `0` to not be a valid sector number. I really do not want to add nil-pointer possibilities here for want of a sentinel value. Can we make `0` invalid? Alternatively, we could make it signed so I can use `-1`, but that would be error prone since the default value will be to replace a sector.

Reply:
If you feel really strongly about `0` being a valid sector number, I will add a boolean field here in addition, rather than use a pointer.